spec/spec_helper.rb

```diff
@@ -45,3 +45,45 @@ def domain_file
     MiqAeDatastore.reset
   end
 end
+
+module AutomateClassDefinitionHook
+  mattr_accessor :hooking
+
+  def self.hooked
+    @hooked ||= []
+  end
+
+  def inherited(other)
+    AutomateClassDefinitionHook.hook(other)
+    super
+  end
+
+  def self.require_with_hook(file_name)
+    self.hooking = true
+    require(file_name).tap { unhook }
+  end
+
+  class TestLoadStub
+    def main
+    end
+  end
+
+  def self.hook(klass)
+    return unless hooking
+    if klass.name&.start_with?("ManageIQ::Automate::")
+      hooked << klass
+      klass.define_singleton_method(:new) do |*_args|
+        AutomateClassDefinitionHook::TestLoadStub.new
+      end
+    end
+  end
+
+  def self.unhook
+    hooked.delete_if do |klass|
+      klass.singleton_class.redefine_method(:new) { |*args| super(*args) }
```

Comment:
Prefer `remove_method` instead of `redefine_method` to call super.  This way it goes back to the original implementation.